core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,97 @@
+package ocr2keeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	httypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/headtracker/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	headBroadcaster    httypes.HeadBroadcaster
+	headCh             chan BlockKey
+	unsubscribe        func()
+	// chDone             chan struct{}
+	// lggr               logger.Logger
+}
+
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster) *AutomationCustomTelemetryService {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		headBroadcaster:    hb,
+		headCh:             make(chan BlockKey), // Channel Size matter?
+		// chDone:             done,
+		// lggr:               lggr,
+	}
+}
+
+// Start starts
+func (e *AutomationCustomTelemetryService) Start(context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		// read from head channel
+		//
+		_, e.unsubscribe = e.headBroadcaster.Subscribe(&headWrapper{e.headCh})
+		go func() {
+			// e.lggr.Infof("Started enhanced telemetry service for job %d", e.job.ID)
+			for blockKey := range e.headCh {
+				// marshall protobuf message to bytes
+				// proto.Marshal takes in a pointer to proto message struct
+				blockNumMsg := &telem.BlockNumber{
+					NodeId:      "faa",
+					Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+					BlockNumber: uint64(blockKey.block),
+					BlockHash:   blockKey.hash,
+				}
+				wrappedMessage := &telem.AutomationTelemWrapper{
+					Msg: &telem.AutomationTelemWrapper_BlockNumber{
+						BlockNumber: blockNumMsg,
+					},
+				}
+				bytes, err := proto.Marshal(wrappedMessage)
+				if err != nil {
+					fmt.Printf("Error occured: %v", err)
+				}
+				e.monitoringEndpoint.SendLog(bytes)
+				// case <-e.chDone:
+				// 	return
+			}
+		}()
+		return nil
+	})
+}
+
+func (e *AutomationCustomTelemetryService) Close() error {
```

Comment:
you can reference this Close func [here](https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr/contract_tracker.go#L171).

basically make sure you unsubscribe and close channels so that we don't have open channels and memory leak